java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceSameLineAsNextBlockCheck.java

```diff
@@ -20,18 +20,30 @@
 package org.sonar.java.checks;
 
 import org.sonar.check.Rule;
+import org.sonar.java.checks.helpers.QuickFixHelper;
+import org.sonar.java.reporting.JavaQuickFix;
+import org.sonar.java.reporting.JavaTextEdit;
 import org.sonar.plugins.java.api.tree.BlockTree;
 import org.sonar.plugins.java.api.tree.SyntaxToken;
 import org.sonarsource.analyzer.commons.annotations.DeprecatedRuleKey;
 
+import static org.sonar.java.reporting.AnalyzerMessage.textSpanBetween;
+
 @DeprecatedRuleKey(ruleKey = "RightCurlyBraceSameLineAsNextBlockCheck", repositoryKey = "squid")
 @Rule(key = "S1107")
 public class RightCurlyBraceSameLineAsNextBlockCheck extends RightCurlyBraceToNextBlockAbstractVisitor {
 
   @Override
   protected void checkTokenPosition(SyntaxToken syntaxToken, BlockTree previousBlock) {
     if (syntaxToken.line() != previousBlock.closeBraceToken().line()) {
-      reportIssue(syntaxToken, "Move this \"" + syntaxToken.text() + "\" on the same line that the previous closing curly brace.");
+      QuickFixHelper.newIssue(context)
+        .forRule(this)
+        .onTree(syntaxToken)
+        .withMessage("Move this \"%s\" on the same line that the previous closing curly brace.", syntaxToken.text())
+        .withQuickFix(() -> JavaQuickFix.newQuickFix("Move to the same line as the closing curly brace")
+          .addTextEdit(JavaTextEdit.replaceTextSpan(textSpanBetween(previousBlock, false, syntaxToken, false), " ")
```

Comment:
What happens if there is a comment between the closing curly brace and the next block?

Reply:
As discussed, the comment will be dropped. In fact, this behavior is the same in all rules where we remove a range, if a comment is present, it will be removed. We might want to do something about it in the future, but for now, I think it's fine, the user can anyway revert the change if needed.